### PR TITLE
update to 1.18

### PR DESCRIPTION
**Pull request by DragonEggBedrockBreaking**
This is currently not ready for merge. It will require maintenance until full release, and is difficult to build with neither sodium's  not reeses sodium options' 1.18 forks being on a maven. Thus, this is marked as a draft.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -3,8 +3,8 @@ plugins {
     id 'maven-publish'
 }
 
-sourceCompatibility = JavaVersion.VERSION_16
-targetCompatibility = JavaVersion.VERSION_16
+sourceCompatibility = "17"
+targetCompatibility = "17"
```
Comment (FlashyReese) [inline build.gradle]:
```suggestion
targetCompatibility = JavaVersion.VERSION_17
```
Just a minor nitpick

**Review by Kichura — COMMENTED** [review]
PSA via line 39 - 40 can be ignored as that loom version is extremely outdated; has also been cut by sodium and other mod projects.

File: build.gradle [hunk 2]
```diff
@@ -4,7 +4,7 @@ plugins {
 }
 
 sourceCompatibility = "17"
```
Comment (Kichura) [inline build.gradle]: Should reflect targetCompatibility's context for consistency, otherwise it looks okay so far.

File: gradle.properties [hunk 3]
```diff
@@ -2,16 +2,16 @@
...
     mod_version=0.3.6
     maven_group=me.flashyreese.mods
     archives_base_name=sodium-extra
 
 # Dependencies
-    reeses_sodium_options=mc1.17.1-1.2.0
-    sodium_version=mc1.17.1-0.3.3
-    fabric_version=0.38.1+1.17
+    reeses_sodium_options=mc1.18-1.2.4
+    sodium_version=mc1.18-0.4.0-alpha1
```
Comment (DragonEggBedrockBreaking) [inline gradle.properties:16]:
```suggestion
    sodium_version=mc1.18-0.4.0-alpha5
```

File: src/main/resources/fabric.mod.json [hunk 4]
```diff
@@ -36,12 +36,11 @@
     "sodium-extra.mixins.json"
   ],
   "depends": {
-    "fabricloader": ">=0.11.3",
+    "fabricloader": ">=0.12.6",
     "fabric": "*",
-    "minecraft": "~1.17",
-    "sodium": "0.3.x"
+    "sodium": "0.4.0-alpha1"
```
Comment (DragonEggBedrockBreaking) [inline src/main/resources/fabric.mod.json:41]:
```suggestion
    “sodium”: “0.4.0-alpha5”
```

**Review by DragonEggBedrockBreaking — COMMENTED** [review]
Correct sodium version

**Comment by Kichura** [comment]
Perhaps add credit of yourself and me for contributing or is that optionally up to reese?

**Comment by DragonEggBedrockBreaking** [comment]
I’d leave it up to reese

**Pull request merged**